File: python/kindergarten-garden/kindergarten_garden.py
class Garden(object):
    def __init__(self, diagram, students = ["Alice", "Bob", "Charlie", "David", "Eve", "Fred", "Ginny", "Harriet", "Ileana", "Joseph", "Kincaid", "Larry"]):
        self.plantCode = {
            "C": "Clover", 
            "G": "Grass",
            "R": "Radishes",
            "V": "Violets"
        }

        self.students = sorted(students)
        self.rows = diagram.split("\n", 2)

    def plants(self, student):
        try:            
            index = self.students.index(student)
            plants = []
            for j in range(2):
                for i in range(index * 2, (index + 1) * 2):
                    try:
                        code = self.rows[j][i]
                        plants.append(self.plantCode[code])
                    except:
                        raise Exception(f"{code} does not match a known plant.")
            return plants
        except:
            raise Exception(f"Student {student} was not found in the class role.")

File: javascript/kindergarten-garden/kindergarten-garden.js
//
// This is only a SKELETON file for the 'Kindergarten Garden' exercise.
// It's been provided as a convenience to get you started writing code faster.
//

const DEFAULT_STUDENTS = [
  'Alice',
  'Bob',
  'Charlie',
  'David',
  'Eve',
  'Fred',
  'Ginny',
  'Harriet',
  'Ileana',
  'Joseph',
  'Kincaid',
  'Larry',
];

const PLANT_CODES = {
  G: 'grass',
  V: 'violets',
  R: 'radishes',
  C: 'clover',
};

export class Garden {
  constructor(diagram, students = DEFAULT_STUDENTS) {
    this.rows = diagram.split('\n')
    this.students = students.sort()
  }

  plants(student) {    
    let index = this.students.indexOf(student)
    if (index < 0)
      throw new Error(`${student} is not in this class.`)

    return this.rows
      .map((row) => row.slice(index * 2, (index + 1) * 2))
      .join("")
      .split('')
      .map((ch) => PLANT_CODES[ch])
  }
}
